pkg/asset/machines/master_test.go

```diff
@@ -453,3 +454,90 @@ func TestFencingCredentialsSecretsFile(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportedSNOPlatforms(t *testing.T) {
+	cases := []struct {
+		name                string
+		platform            types.Platform
+		bootstrapInPlace    *types.BootstrapInPlace
+		machinePoolPlatform types.MachinePoolPlatform
+		expectedError       bool
+	}{
+		{
+			name: "AWS platform",
+			platform: types.Platform{
+				AWS: &awstypes.Platform{
+					Region: "us-east-1",
+					DefaultMachinePlatform: &awstypes.MachinePool{
+						InstanceType: "TEST_INSTANCE_TYPE",
+					},
+				},
+			},
+			bootstrapInPlace: &types.BootstrapInPlace{},
+			machinePoolPlatform: types.MachinePoolPlatform{
+				AWS: &awstypes.MachinePool{
+					Zones: []string{"us-east-1a"},
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "None platform",
+			platform: types.Platform{
+				None: &none.Platform{},
+			},
+			machinePoolPlatform: types.MachinePoolPlatform{},
+			expectedError:       false,
+		},
+		{
+			name: "IBMCloud platform",
+			platform: types.Platform{
+				IBMCloud: &ibmcloudtypes.Platform{},
+			},
+			bootstrapInPlace: &types.BootstrapInPlace{},
+			machinePoolPlatform: types.MachinePoolPlatform{
+				IBMCloud: &ibmcloudtypes.MachinePool{},
+			},
+			expectedError: false,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			parents := asset.Parents{}
+			parents.Add(
+				&installconfig.ClusterID{
+					UUID:    "test-uuid",
+					InfraID: "test-infra-id",
+				},
+				installconfig.MakeAsset(
+					&types.InstallConfig{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-cluster",
+						},
+						SSHKey:     "ssh-rsa: dummy-key",
+						BaseDomain: "test-domain",
+						Platform:   tc.platform,
+						ControlPlane: &types.MachinePool{
+							Hyperthreading: types.HyperthreadingEnabled,
+							Replicas:       pointer.Int64Ptr(1),
+							Platform:       tc.machinePoolPlatform,
+						},
+					}),
```

Comment:
`bootstrapInPlace` should be set here for the install-config, right?

```
BootstrapInPlace: tc.bootstrapInPlace,
```

Reply:
Oh sorry, I saw it set in the test cases, but not being used anywhere when running the tests (i.e. that is not used in the mock install-config).